src/frontend/src/optimizer/mod.rs

```diff
@@ -285,13 +319,16 @@ impl PlanRoot {
             ctx.trace(plan.explain_to_string());
         }
 
-        Ok(plan)
+        self.plan = plan;
+        self.phase = PlanPhase::Batch;
+        Ok(self.plan.clone())
     }
 
     /// Optimize and generate a batch query plan for distributed execution.
-    pub fn gen_batch_distributed_plan(&mut self, batch_plan: PlanRef) -> Result<PlanRef> {
-        self.set_required_dist(RequiredDist::single());
-        let mut plan = batch_plan;
+    pub fn gen_batch_distributed_plan(mut self) -> Result<PlanRef> {
+        assert_eq!(self.phase, PlanPhase::Batch);
+        self.required_dist = RequiredDist::single();
+        let mut plan = self.plan;
```

Comment:
`gen_batch_distributed_plan` will change `required_dist` of the `PlanRoot`, so once we call this method, we can't use it to generate another plan anymore, but the planner test did that.